### PR TITLE
Upgrade dependencies, bumps substrate to 2.0.1

### LINKED ISSUE #218: update substrate to `2.0.1`
The motivation is to avoid pinning an old version of the nightly compiler to make substrate to compile which is annoying.

### PR DESCRIPTION
**Pull request by ascjones**
Fixes #218

### CODE REVIEW

**Comment by ascjones** [comment]
Fix for failure in multi-addr, might need to wait for new libp2p release? https://github.com/libp2p/rust-libp2p/commit/aa2547ef6d91e675ca1ef89952011e4ad78aa4b4

**Comment by niklasad1** [comment]
Fix for failure in multi-addr, possibly need to wait for release of https://crates.io/crates/sc-network which depends on the old libp2p

> It is getting backported https://github.com/libp2p/rust-libp2p/pull/1925

**Comment by mxinden** [comment]
> > Fix for failure in multi-addr, possibly need to wait for release of https://crates.io/crates/sc-network which depends on the old libp2p
> 
> It is getting backported [libp2p/rust-libp2p#1925](https://github.com/libp2p/rust-libp2p/pull/1925)

Backport is released. Please update to `parity-multiaddr` `v0.9.7`.

**Comment by ascjones** [comment]
Patched `parity-multiaddr` is a transient dependency so it now builds with no change.

File: Cargo.toml [hunk 1]
```diff
@@ -24,39 +24,39 @@ client = ["substrate-subxt-client"]
...
-sp-runtime = { version = "2.0.0", package = "sp-runtime" }
-sp-version = { version = "2.0.0", package = "sp-version" }
-pallet-indices = { version = "2.0.0", package = "pallet-indices" }
+num-traits = { version = "0.2.14", default-features = false }
+serde = { version = "1.0.119", features = ["derive"] }
+serde_json = "1.0.61"
+url = "2.2.0"
+codec = { package = "parity-scale-codec", version = "1.3.6", default-features = false, features = ["derive", "full"] }
+
+frame-metadata = { version = "12.0.1", package = "frame-metadata" }
```
Comment (niklasad1) [inline Cargo.toml:37]: not really introduced by this PR but can we remove the redundant "package" here and below?

```suggestion
frame-metadata = "12.0.1"
```

Comment (ascjones) [reply Cargo.toml]: Sure, probably a hangover from when git deps used

File: client/Cargo.toml [hunk 2]
```diff
@@ -12,20 +12,20 @@ description = "Embed a substrate node into your subxt application."
...
+sc-network = { version = "0.8.1", default-features = false }
+sc-service = { version = "0.8.1", default-features = false }
+serde_json = "1.0.61"
+sp-keyring = "2.0.1"
+thiserror = "1.0.23"
 
 [dev-dependencies]
-async-std = { version = "1.6.4", features = ["attributes"] }
-env_logger = "0.7.1"
+async-std = { version = "1.8.0", features = ["attributes"] }
```
Comment (niklasad1) [inline client/Cargo.toml:27]: NOTE, this feature will be leaked into `dependencies` (it's fine but still.. the new feature resolver[https://github.com/rust-lang/cargo/pull/8997) is stabilized and will fix this FWIW)

**Review by niklasad1 — APPROVED** [review]
great, LGTM

**Pull request merged**